File: mobile_app/the_store_app/lib/entity/shop/shop_dto.dart

import 'package:freezed_annotation/freezed_annotation.dart';
part 'shop_dto.freezed.dart';
part 'shop_dto.g.dart';

@Freezed(
  fromJson: true,
  toJson: true,
)
class ShopDto with _$ShopDto{
  factory ShopDto({
    required int id,
    required String shopName,
    required String shopPhone,
    required String shopWorkHours,
    required String shopPicture,
    required String lat,
    required String lon,
}) = _ShopDto;

  factory ShopDto.fromJson(Map<String, dynamic> json) => _$ShopDtoFromJson(json);
}

File: backend/src/main/java/ru/fbtw/thestore/backend/data/delivery/dto/ShopDto.java
package ru.fbtw.thestore.backend.data.delivery.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.*;

import java.io.Serializable;

/**
 * A DTO for the {@link ru.fbtw.thestore.backend.domains.delivery.Shop} entity
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ShopDto implements Serializable {
	private Long id;
	@NotBlank(message = "Phone number is required")
	private String shopName;
	@NotBlank(message = "Phone number is required")
	private String shopAddress;
	@Pattern(regexp = "^\\+7\\d{10}$",
			message = "Phone number must be in the format +79999999999")
	@NotBlank(message = "Phone number is required")
	private String shopPhone;
	@NotBlank(message = "Phone number is required")
	private String shopWorkHours;
	@NotBlank(message = "Phone number is required")
	private String shopPicture;
	@NotBlank(message = "Phone number is required")
	private String lat;
	@NotBlank(message = "Phone number is required")
	private String lon;

}